backend/src/main/java/com/openelements/maven/initializer/backend/service/ProjectGeneratorService.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.openelements.maven.initializer.backend.service;
+
+import com.openelements.maven.initializer.backend.dto.ProjectRequestDTO;
+import eu.maveniverse.maven.toolbox.shared.ToolboxCommando;
+import eu.maveniverse.maven.toolbox.shared.internal.PomSuppliers;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProjectGeneratorService {
+
+  private static final Logger logger = LoggerFactory.getLogger(ProjectGeneratorService.class);
+  private final ToolboxCommando toolboxCommando;
+  private final ProjectZipperService zipperService;
+  private final ProjectStructureService structureService;
+
+  private static final List<String> DEFAULT_DEPENDENCIES = List.of("commons-io:commons-io:2.15.1");
+
+  private static final List<String> DEFAULT_PLUGINS =
+      List.of(
+          "org.apache.maven.plugins:maven-compiler-plugin:3.14.0",
```

Comment:
Currently, it is ok. But in the future, we will need a mechanism that checks the current version of the plugins. I will create a new issue for that.